### DISCUSSION TITLE
Launching Executable opens Minimized Application.

### DISCUSSION THREAD

**Discussion by Sokoloft** [streamlink/streamlink-twitch-gui#867] [Q&A] [answered]
As my title says. I'd like the application to open if the executable is launched again. Is this possible?

I use an application called hotkeyD with a bunch of batch scripts I've made. The script looks like this:

```
cd "C:\Program Files\Streamlink Twitch GUI\"
start streamlink-twitch-gui.exe
exit
``` 

Even some argument like "-max" would be great and would work for my usage case. I've tried the /max argument for start but it does nothing.

**Comment by bastimeyer** [comment, marked as answer]
You can see the available parameters on this wiki page:
https://github.com/streamlink/streamlink-twitch-gui/wiki/Parameters

There are however currently no parameters for unminimizing and unmaximizing the application window or giving it focus again if it's not in focus, so this would have to be implemented first.

These window related parameters are however meant as launch parameters, to configure auto-start behavior for example, and not for controlling the app window by launching secondary instances. That should be done via the system's window manager, especially when setting up hotkeys. At least that's how it's done in the non-Windows world.

> `-max`
> `/max`

These parameter notations are Windows specific and not POSIX compliant.

----

I just noticed that I forgot to update the --launch parameter for the 2.0.0 release. This needs to be fixed.

**Reply by bastimeyer** [reply]
`--min` and `--max` now toggle the application window's minimized and maximized state:
b16d8f592cd959f7c084f0236866a4e5061c7bac

And this is the fix for the `--launch` parameter:
d5d519c401104e745b7f952b8df01dba23adf8bf

The next nightly build will include these changes. Otherwise, build from the master branch or wait for the next stable release.

**Comment by Sokoloft** [comment]
Hello.

I'm afraid this isn't working for me. When streamlink-twitch-gui is minimized it goes to my system tray. When I press my hotkey to open the batch I have. It still stays minimized. I added the --max option to the end of it. Maybe there is something I missed? I tried just making a desktop shortcut and editing its properties adding --max. Same result. 

```
cd "C:\Program Files\Streamlink Twitch GUI\"
streamlink-twitch-gui.exe "--max"
exit
```

I built from source following the instruction you have. Below is a screenshot of the built app.
![streamlink-twitch-gui_LuVBMaXc4r](https://user-images.githubusercontent.com/49052130/153060039-0680fbba-6da6-48e8-8641-af942ffbd115.png)

Thanks so much.

**Reply by bastimeyer** [reply]
> When streamlink-twitch-gui is minimized it goes to my system tray

`--min` and `--max` only affect the minimized and maximized state of the app window. `--max` is not the opposite of `--min`, both are different things.

As I've said, there is currently no parameter for restoring the entire window state. Resetting the tray icon state is a bit more complicated due to user settings being involved here and how the initial tray-icon state gets set up. Implementing a toggle for that is something I think is not worth spending any time on, because it would require rewriting other stuff.

**Comment by Sokoloft** [comment]
I should also clarify. I want streamlink-twitch-gui to pop up when I press my hotkey. If you exec the .exe while its already open it does nothing. It'd be great if it'd pop up if that's possible.
